fl4health/clients/basic_client.py

```diff
@@ -138,9 +138,15 @@ def get_parameters(self, config: Config) -> NDArrays:
             NDArrays: These are the parameters to be sent to the server. At minimum they represent the relevant model
                 parameters to be aggregated, but can contain more information.
         """
-
-        assert self.model is not None and self.parameter_exchanger is not None
-        return self.parameter_exchanger.push_parameters(self.model, config=config)
+        if not self.initialized:
+            # If initialized==False, we are doing client side initialization (get_parameters called before fit)
```

Comment:
I'd suggest fleshing this out slightly more to something like
```
If initialized==False, the server is requesting model parameters from which to initialize all other clients. As such get_parameters is being called before fit or evaluate, so we must call setup_client first.
```

Reply:
Thanks, this is much better :)